lib/airports.rb

```diff
@@ -11,10 +11,24 @@ def self.find_by_iata_code(iata_code)
       new(airport_data.each_with_object({}) { |(k,v), h| h[k.to_sym] = v })
   end
 
+  def self.find_by_icao_code(icao_code)
+    return unless icao_code.length == 4
```

Comment:
I think I'd be inclined to take an approach more like this:

```ruby
def self.find_by_icao_code(icao_code)
  return unless icao_code.length == 4

  airport_data = parsed_data.find do |_, data|
    data['icao'] == icao_code
  end

  return unless airport_data

  Airport.
    new(airport_data.each_with_object({}) { |(k,v), h| h[k.to_sym] = v })
end
```

What do you think?

Reply:
That looks like a more elegant solution. Wouldn't have thought about doing it that way.